Add new syringe sprites.

UnokiAs opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
Adding new sprites for syringes for Normal,Piercing and BlueSpace.  Because all the syringes types look the same.
~~(For some reason the sprites don't show in game any help are welcome)~~ Fixed
## Why It's Good For The Game
It can be quite difficult to differentiate betwen each syringe type, I also change the normal syringe to look like a plastique syringe to be easier to look at it when it's in a box or on the ground
![Capture d’écran 2021-04-22 224051](https://user-images.githubusercontent.com/82665345/115782325-d7394a80-a3bb-11eb-8813-87944712ffc0.png)


https://user-images.githubusercontent.com/82665345/115958028-e9bf9b00-a505-11eb-89c9-9167b5000035.mp4


<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: Unoki
imageadd: Added new syringe sprites (Normal,Piercing and BlueSpace.)
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

UnokiAs committed:
Add new syringe sprite.

optimumtact commented:
As mentioned in discord you need to set the icon_state for each type to be the zero state in each type definition
icon_state="bluespace_0"

etc

right now they're defaulting to "0" and that state no longer exists.

CRITAWAKETS commented:
you should make the bluespace syringe thicker to emphasize that its a big boy and probably wont fit in your syringe gun

UnokiAs commented:
> you should make the bluespace syringe thicker to emphasize that its a big boy and probably wont fit in your syringe gun

Hum I didn't do it because the needle would look weird but I will try something yeah

cacogen commented:
I think there's a stray white pixel in the pulsing animation

UnokiAs commented:
> I think there's a stray white pixel in the pulsing animation

Oh on every of them or one in particular ? Is it on the syringe or outside of it ? Thank you in advance !

cacogen commented:
https://github.com/tgstation/tgstation/pull/58619/checks?check_run_id=2414273261

I wasn't sure if it was on my end from it not downloading properly or something but for me at least in a few frames of the bluespace syringe animations there's a white pixel

![](https://i.imgur.com/4eDwISd.png|)

UnokiAs commented:
> https://github.com/tgstation/tgstation/pull/58619/checks?check_run_id=2414273261
> 
> I wasn't sure if it was on my end from it not downloading properly or something but for me at least in a few frames of the bluespace syringe animations there's a white pixel
> 
> ![](https://camo.githubusercontent.com/4f5f8a3dd3fd023419459782c1269045b73c0083e9d57978953f9adc2b485efa/68747470733a2f2f692e696d6775722e636f6d2f346544774953642e706e67253743)

Thank you ! I will check and fix it 😄

Fikou commented on code/modules/reagents/reagent_containers/syringes.dm:
```diff
@@ -138,7 +138,7 @@
 /obj/item/reagent_containers/syringe/update_icon_state()
 	var/rounded_vol = get_rounded_vol()
-	icon_state = "[rounded_vol]"
+	icon_state = "[base_icon_state]_[rounded_vol]"
```
base_icon_state need to be "syringe" on the regular syringes

UnokiAs replied:
Done thanks for the help

UnokiAs committed:
Update syringes.dm

UnokiAs committed:
Edited the syringe projectiles to match the new syringe.

UnokiAs commented:
Alright I also changed  syringe projectile to match with the new sprite and fixed the normal syringe being not center also for the stray white pixel I didn't found it so I guess it's IconDiffbot being funky.

Fikou reviewed (changes requested):
You should probably sprite the inhands for the syringes and in syringe's Initialize proc do AddElement(/datum/element/update_icon_updates_onmob)
so that when the icon state changes it updates the inhands (this should be in the current version of the code too, for some reason nobody added it)
Other than that, looks good.

UnokiAs committed:
Inhand (Work in progress)

Still have issue with the placement but I will fix it any help is welcome. (For the back syringe showing)

UnokiAs commented:
So yeah I'm working on the inhand but I think the placement is wrong:
![Capture d’écran 2021-04-23 224254](https://user-images.githubusercontent.com/82665345/115928390-5004d900-a486-11eb-8e8b-d9445150b3cf.png)
![Capture d’écran 2021-04-23 224241](https://user-images.githubusercontent.com/82665345/115928393-509d6f80-a486-11eb-9f41-337b449a118d.png)
![Capture d’écran 2021-04-23 224321](https://user-images.githubusercontent.com/82665345/115928394-509d6f80-a486-11eb-97c8-046920658d97.png)
![Capture d’écran 2021-04-23 224308](https://user-images.githubusercontent.com/82665345/115928395-51360600-a486-11eb-972f-70dc649bab78.png)

UnokiAs committed:
Inhand syringe fix

UnokiAs committed:
Inhand syringe fix 2

UnokiAs committed:
Inhand syringe fix 3 Final... MAYBE.

UnokiAs commented:
Alright INHAND syringe done !

https://user-images.githubusercontent.com/82665345/115958010-d280ad80-a505-11eb-8b50-d2225bd4e360.mp4

Fikou reviewed (approved):
Looks good 👍

UnokiAs committed:
Update medical_righthand.dmi

Fixing piercing__0 <---- pixel off issue

UnokiAs commented:
Oh gaming ! 
Thanks to 
Sparkezel, Fikou and oranges for the coding help.
Twaticus for inhand sprite help.
😄
